elixir/apps/domain/lib/domain/resources/resource/changeset.ex

```diff
@@ -5,7 +5,7 @@ defmodule Domain.Resources.Resource.Changeset do
 
   @fields ~w[address address_description name type]a
   @update_fields ~w[name address_description]a
-  @required_fields ~w[name address address_description type]a
+  @required_fields ~w[name address type]a
```

Comment:
Should we keep it required?

Reply:
The Clients already handle the case where it's empty. I think we may want to rename the UI label to `Display Address` down the line, or use its emptiness to hide the Resource in the Clients (this has been requested a few times)

Reply:
So the logic in the Clients is, "if address_description exists, display that, otherwise display the address". The idea was that an admin can override what's displayed/copyable in the Resource menu.

Reply:
<img width="411" alt="Screenshot 2024-06-06 at 5 15 13 PM" src="https://github.com/firezone/firezone/assets/167144/df3ffa82-5275-48e5-a2bc-14edc4c19491">

`address_description` is probably not the best name for this field... maybe `display_address` is better.

Reply:
`display_address` might be not good too because we should support something like markdown in there and it's not just the address. Maybe we can do something like `client_menu_description`?

The only reason why I was thinking to make it required is that our UI will auto-fill that field and I would rather have more logic on the portal (how it should be filled) than logic on how to fallback to something on the client when the field is empty.

Reply:
Maybe we can search the DB and see how many resources have `name = address` to see if users are having to change them

Reply:
Hm, I see two good solutions here. One is to select all text in an auto-filled input when it's focused, so you can quickly erase if we did not guess it right. Another one is not to auto-fille. But auto-filling name (which is hard to guess) is less effective than auto-filling the address description, where it's usually just an `http://` + address.

Reply:
I don't think it's safe to assume address description will be a URL more than 50% of the time. If we guess incorrectly for even 1/10, the admin needs to remember each time they create a Resource to change it.

Most of the time text inputs are not auto populated on the internet, so the admin essentially needs to learn this weird quirk of our product in order to trust and make use of it.

I just think for the vast majority of cases, the reaction is going to be surprise and minor annoyance rather than relief the field is filled out, saving them a few keystrokes, right?

Reply:
Yeah, we can drop auto-fill, I'm not strongly about it. The only thing I really don't like is to have fallback logic on the clients and nullable field in DB that can be used to remove that logic

Reply:
Hmmm, yeah, I agree. Let's agree to revisit this some more in the next iteration of Client UX changes.

I understand now part of the reason is to avoid column nulls